cirq-superstaq/cirq_superstaq/service_test.py

```diff
@@ -63,6 +63,21 @@ def test_counts_to_results() -> None:
     result = css.service.counts_to_results({"00": 50, "11": 50}, circuit, cirq.ParamResolver({}))
     assert result.histogram(key="01") == collections.Counter({0: 50, 3: 50})
 
+    result = css.service.counts_to_results(
+        {"00": 50.0, "11": 50.0}, circuit, cirq.ParamResolver({})
+    )
+    assert result.histogram(key="01") == collections.Counter({0: 50, 3: 50})
+
+    with pytest.warns(UserWarning, match="raw counts contain fractional"):
+        result = css.service.counts_to_results(
+            {"00": 50.1, "11": 49.9}, circuit, cirq.ParamResolver({})
+        )
+
+    with pytest.warns(UserWarning, match="raw counts contain negative"):
+        result = css.service.counts_to_results(
+            {"00": -50.1, "11": 99.9}, circuit, cirq.ParamResolver({})
+        )
+
```

Comment:
Could you add asserts for the resulting counts to these two? Just to make it a bit more explicit about what the code does with these unexpected numbers.

Reply:
Thanks for this -- it uncovered another error. We can't handle negative numbers in the `cirq.ResultDict` for the same reason we can't handle fractional numbers, because you can't add negatively-many elements to an array:
```python
# In counts_to_results:
for key, counts_of_key in counter.items():
        keys_as_list: List[int] = list(map(int, key))

        counts_of_key = round(counts_of_key)
        for _ in range(counts_of_key):
            samples.append(keys_as_list)

# counter = {"00": 2, "11": 3} -> samples = [[0, 0], [0, 0], [1, 1], [1, 1], [1, 1]]

# counter = {"00": -2, "11": 3} -> samples = [[1, 1], [1, 1], [1, 1]]
```

`cirq.ResultDict` needs `samples` for its `measurements` parameter, so I don't see how we can get around this without just ignoring negative counts... (which is maybe fine since we have the warning)

Reply:
^ https://github.com/Infleqtion/client-superstaq/issues/785